### PR TITLE
Bump revision in DNS watcher of ZK DNS

### PR DESCRIPTION
**Pull request by noggi**
My previous PR (https://github.com/airbnb/synapse/pull/236) only fixed the ZK 
part of ZK DNS watcher. As the result, only ZK updates were handled correctly.

This change fixes the DNS part as well, so IP address updates will now also
trigger haproxy reconfiguration.

@jimmyngo @tuckerman @jolynch

### CODE REVIEW

**Comment by tuckerman** [comment]
LGTM

**Comment by jimmyngo** [comment]
LGTM!

File: lib/synapse/service_watcher/zookeeper_dns.rb [hunk 1]
```diff
@@ -51,8 +51,9 @@ class Dns < Synapse::ServiceWatcher::DnsWatcher
       # Overrides the discovery_servers method on the parent class
       attr_accessor :discovery_servers
 
-      def initialize(opts={}, synapse, message_queue)
+      def initialize(opts={}, parent=nil, synapse, message_queue)
```
Comment (tuckerman) [inline lib/synapse/service_watcher/zookeeper_dns.rb]: is this instantiated anywhere other than your change below? if not, why the need for the default arg?

**Comment by jolynch** [comment]
Thanks for the fix :-)

**Pull request merged**